src/WorkbenchAccessManager.php

```diff
@@ -403,4 +433,39 @@ public function flushFields() {
     drupal_set_message($this->t('Field settings reset.'));
   }
 
+  /**
+   * Loads the saved role sections for a given role ID.
+   *
+   * @param string $role_id
+   *   The role ID.
+   *
+   * @return array
+   *   Sections for role.
+   */
+  protected function loadRoleSections($role_id) {
```

Comment:
Perfectly happy to do it as part of the breakout into services, as it makes a lot of sense.

Roles use state because it was expedient, and I didn't want to manage custom database tables.